src/storm/mesos/MesosNimbus.java

```diff
@@ -707,7 +751,58 @@ public void executorLost(SchedulerDriver driver, ExecutorID executor, SlaveID sl
       LOG.info("Executor lost: executor=" + executor + " slave=" + slave);
     }
   }
-
+    private FrameworkInfo.Builder createFrameworkBuilder() throws IOException {
+        
+        String id = (String)getLocalState().get(FRAMEWORK_ID);
+        Number failoverTimeout = Optional.fromNullable((Number) getConfig().get(CONF_MASTER_FAILOVER_TIMEOUT_SECS)).or(3600);
+        String role = Optional.fromNullable((String) getConfig().get(CONF_MESOS_ROLE)).or("*");     
+        Boolean checkpoint = Optional.fromNullable((Boolean)getConfig().get(CONF_MESOS_CHECKPOINT)).or(false);
+        String framework_name = Optional.fromNullable((String)getConfig().get(CONF_MESOS_FRAMEWORK_NAME)).or("Storm!!!");
+
+        FrameworkInfo.Builder finfo = FrameworkInfo.newBuilder()
+            .setName(framework_name)
+            .setFailoverTimeout(failoverTimeout.doubleValue())
+            .setUser("")
+            .setRole(role)
+            .setCheckpoint(checkpoint);
+
+        if (id != null) {
+          finfo.setId(FrameworkID.newBuilder().setValue(id).build());
+        } 
+        
+        return finfo;
+    }
+    
+    private Credential checkForAuthentication(FrameworkInfo.Builder finfo) {
```

Comment:
I will do that.

Thanks.

On Fri, May 1, 2015 at 5:30 PM, Timothy Chen <notifications@github.com>
wrote:

> In src/storm/mesos/MesosNimbus.java
> <https://github.com/mesos/storm/pull/37#discussion_r29535256>:
>
> > +
> > +        FrameworkInfo.Builder finfo = FrameworkInfo.newBuilder()
> > +            .setName(framework_name)
> > +            .setFailoverTimeout(failoverTimeout.doubleValue())
> > +            .setUser("")
> > +            .setRole(role)
> > +            .setCheckpoint(checkpoint);
> > +
> > +        if (id != null) {
> > +          finfo.setId(FrameworkID.newBuilder().setValue(id).build());
> > +        }
> > +
> > +        return finfo;
> > +    }
> > +
> > +    private Credential checkForAuthentication(FrameworkInfo.Builder finfo) {
>
> the name checkForAuthentication doesn't seems to suggest it's returning
> Crendential. How about calling it getAuthentication that returns a Optional?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/storm/pull/37/files#r29535256>.
>